# Add or change Solaris-specific file and directory names.
# This change was implemented in-house and is not suitable for upstream use.
#
--- src/include/tcsd.h	2014-04-24 11:05:44.000000000 -0700
+++ src/include/tcsd.h	2016-04-22 11:29:09.043084190 -0700
@@ -50,15 +50,27 @@
 	int disable_ipv6;
 };
 
+#ifdef SOLARIS
+#define	TCSD_DEFAULT_CONFIG_FILE	"/etc/security/tcsd.conf"
+#else
 #define TCSD_DEFAULT_CONFIG_FILE	ETC_PREFIX "/tcsd.conf"
+#endif
 extern char *tcsd_config_file;
 
 #define TSS_USER_NAME		"tss"
 #define TSS_GROUP_NAME		"tss"
 
 #define TCSD_DEFAULT_MAX_THREADS	10
+#ifdef SOLARIS
+#define	TCSD_DEFAULT_SYSTEM_PS_DIR	"/var/tpm/system"
+#define	TCSD_DEFAULT_SYSTEM_PS_FILE	"/var/tpm/system/system.data"
+#define	TCSD_DEFAULT_SOCKET		"/var/tpm/tcsd-socket"
+#else
 #define TCSD_DEFAULT_SYSTEM_PS_FILE	VAR_PREFIX "/lib/tpm/system.data"
 #define TCSD_DEFAULT_SYSTEM_PS_DIR	VAR_PREFIX "/lib/tpm"
+#define	TCSD_DEFAULT_SOCKET		VAR_PREFIX "/run/tpm/tpmd_socket:0"
+
+#endif	/* SOLARIS */
 #define TCSD_DEFAULT_FIRMWARE_LOG_FILE	"/sys/kernel/security/tpm0/binary_bios_measurements"
 #define TCSD_DEFAULT_KERNEL_LOG_FILE	"/sys/kernel/security/ima/binary_runtime_measurements"
 #define TCSD_DEFAULT_FIRMWARE_PCRS	0x00000000
